src/libostree/ostree-sysroot.c

```diff
@@ -1545,7 +1545,7 @@ ostree_sysroot_simple_write_deployment (OstreeSysroot      *sysroot,
           g_ptr_array_add (new_deployments, g_object_ref (deployment));
         }
 
-      if (!added_new)
+      if ((!added_new) && is_merge_or_booted)
```

Comment:
To be honest, I don't plan to use this flag. I just stumbled upon the inconsistency between the code and the comment.
This suggested change makes the code behave as described in the comment, the way I understand it.

I don't have an opinion about the desirable behavior.